vignettes/r-notebook.Rmd

```diff
@@ -0,0 +1,129 @@
+---
+title: "R Notebook"
+author: "Kevin Ushey"
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{R Notebook}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+This vignette describes the HTML notebook format, and is primarily intended for
+front-end applications using or embedding R, or other users who are interested
+in understanding and parsing the R Notebook format.
+
+R Notebooks are HTML documents with data written and encoded in such a way that:
+
+1. The generating `.Rmd` document can be recovered, and
+2. Chunk outputs can be recovered.
+
+To generate an R Notebook, you can use `rmarkdown::render()` and specify the
+`rmarkdown::html_notebook` output format in your document's YAML metadata.
+Documents rendered in this form will be generated with the `.nb.html` file
+extension, to indicate that they are HTML notebooks.
+
+To ensure chunk outputs can be recovered, the elements of the R Markdown 
+document are enclosed with HTML comments, providing more information on the
+output. For example, chunk output might be serialized as:
+
+```
+<!-- rnb-chunk-begin -->
+<!-- rnb-output-begin -->
+<pre><code>Hello, World!</code></pre>
+<!-- rnb-output-end -->
+<!-- rnb-chunk-end -->
+```
+
+Because R Notebooks are just HTML documents, they can be opened and viewed in 
+any web browser; however, hosting environments can be configured to recover and
+open the underlying `.Rmd` document, and also recover and display chunk outputs
+as appropriate.
+
+## Generating R Notebooks with Custom Output
+
+It's possible to render an HTML notebook with custom chunk outputs inserted in 
+lieu of the result that would be generated by evaluating the associated R code. 
+(This can be useful when 'saving the state' of an `.Rmd` with chunk outputs 
```

Comment:
This can be useful for front-end editors that show output of chunk execution inline or for conversion programs from other notebook formats where output is already available from the source format.